src/wasm-builder.h

```diff
@@ -321,12 +322,8 @@ class Builder {
   static Index addVar(Function* func, Name name, WasmType type) {
     // always ok to add a var, it does not affect other indices
     Index index = func->getNumLocals();
-    if (name.is()) {
-      // if there is a name, apply it, but here we assume all the rest have names too FIXME
-      assert(func->localIndices.size() == func->params.size() + func->vars.size());
-      func->localIndices[name] = index;
-      func->localNames.push_back(name);
-    }
+    if (name.is()) func->localIndices[name] = index;
+    func->localNames.push_back(name);
```

Comment:
this doesn't look right. the previous line checks if name, and so it seems like this line could add a null name to localNames? is the idea to keep the array at the right length? that seems confusing and should at least merit a comment, but i also think nulls in that array are odd, we should avoid that.

but more largely, i think this and the method before it `addParam` should handle missing names identically (instead of one requiring names and the other not). i'd be ok to leave this to a followup though, as I don't know offhand what the best fix is. but my point is that this PR seems to make things worse rather than better - i'm ok with that only if you can commit to the followup soon.

Reply:
(lgtm with that)

Reply:
Yeah, I can clean this up in a follow up. I think I did do this to keep the length of the array correct, which seemed better than assuming all previous variables had also had names.

Reply:
Now that Functions can create default names, perhaps it makes the most sense to require a name for everything and fix any passes that omit names to use default names instead.

Reply:
Cool, thanks. Yeah, that might be a good option. Let's discuss more in the followup. I'll merge this in now.